Indicators/IndicatorDerivativeOscillator.cs

```diff
@@ -0,0 +1,89 @@
+    /// a moving average convergence-divergence (MACD) histogram to a double-smoothed relative strength index (RSI).
+    /// </summary>
+    public class IndicatorDerivativeOscillator : Indicator, IIndicatorWarmUpPeriodProvider
+    {
+        private readonly RelativeStrengthIndex _rsi;
+        private readonly ExponentialMovingAverage _smoothedRsi;
+        private readonly ExponentialMovingAverage _doubleSmoothedRsi;
+        private readonly SimpleMovingAverage _signalLine;
+        private readonly int _r1;
+        private readonly int _a1;
+        private readonly int _a2;
+        private readonly int _a3;
+
+        /// <summary>
+        /// Initializes a new instance of the IndicatorDerivativeOscillator class with the specified name and periods.
+        /// </summary>
+        /// <param name="name">The name of the indicator</param>
+        /// <param name="r1">The period for the RSI calculation</param>
+        /// <param name="a1">The period for the smoothing RSI</param>
+        /// <param name="a2">The period for the double smoothing RSI</param>
+        /// <param name="a3">The period for the signal line</param>
+        public IndicatorDerivativeOscillator(string name, int r1, int a1, int a2, int a3) : base(name)
+        {
+            _r1 = r1;
+            _a1 = a1;
+            _a2 = a2;
+            _a3 = a3;
+            _rsi = new RelativeStrengthIndex($"{name}_RSI", r1);
+            _smoothedRsi = new ExponentialMovingAverage($"{name}_SmoothedRSI", a1);
+            _doubleSmoothedRsi = new ExponentialMovingAverage($"{name}_DoubleSmoothedRSI", a2);
+            _signalLine = new SimpleMovingAverage($"{name}_SignalLine", a3);
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady =>
+            _rsi.IsReady && _smoothedRsi.IsReady && _doubleSmoothedRsi.IsReady && _signalLine.IsReady;
+
+        /// <summary>
+        /// Computes the next value for the derivative oscillator indicator from the given state
+        /// </summary>
+        /// <param name="input">The input value to this indicator on this time step</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IndicatorDataPoint input)
+        {
+            if (!IsReady)
+            {
+                return 0;
+            }
+
+            _rsi.Update(input);
+
+            _smoothedRsi.Update(_rsi.Current);
+
+            _doubleSmoothedRsi.Update(_smoothedRsi.Current);
+
+            _signalLine.Update(_doubleSmoothedRsi.Current);
+
+            return _doubleSmoothedRsi.Current.Value - _signalLine.Current.Value;
+        }
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
```

Comment:
1. Its a good practice to have class properties after fields and before constructors and methos when possible.
2. I think this was never assigned.